### PR TITLE
DM-43269: Refactor StarTracker code

### PR DESCRIPTION
**Pull request by mfisherlevine**

### CODE REVIEW

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 1]
```diff
@@ -0,0 +1,625 @@
...
+)
+
+
+def getStreamingSequences(dayObs):
+    """Get the streaming sequences for a dayObs.
+
+    Note that this will need rewriting very soon once the way the data is
+    organised on disk is changed.
+
+    TODO: Will be moved to summit_extras on DM-43269
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:67]: `TODO` now `Done`!

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 2]
```diff
@@ -0,0 +1,625 @@
+# This file is part of summit_utils.
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:1]: Part of what now?!

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 3]
```diff
@@ -0,0 +1,625 @@
...
+def plotSourcesOnImage(parentFilename, sources):
+    """Plot one of more source on top of an image.
+
+    Parameters
+    ----------
+    parentFilename : `str`
+        The full path to the parent (.tif) file.
+    sources : `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source` or
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:573]: `lsst.summit.utils.astrometry.fastStarTrackerAnalysis` -> `lsst.summit.utils.starTracker` ?

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 4]
```diff
@@ -0,0 +1,625 @@
...
+    """Plot the centroid movements and fluxes etc for a set of results.
+
+    By default the brightest source in each image is plotted, but this can be
+    changed by setting ``sourceIndex`` to values greater than 0 to move through
+    the list of sources in each image.
+
+    Parameters
+    ----------
+    results : `dict` of `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:452]: `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source` -> `lsst.summit.extras.fastStarTrackerAnalysis.Source` ?

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 5]
```diff
@@ -0,0 +1,625 @@
...
+    input images were the same size (because we read out sub frames, and
+    analyzing these with full frame data invalidates the centroid coordinates).
+
+    Also displays the maximum (x, y) movements between adjacent exposures, and
+    the mean and stddev of the main source's flux.
+
+    Parameters
+    ----------
+    results : `dict` of `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:348]: `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source` -> `lsst.summit.extras.fastStarTrackerAnalysis.Source` ?

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 6]
```diff
@@ -0,0 +1,625 @@
...
+        return np.nan
+
+
+def findFastStarTrackerImageSources(filename, boxSize, attachCutouts=True):
+    """Analyze a single FastStarTracker image.
+
+    Parameters
+    ----------
+    filename : `str`
+        The full
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:280]: Don't leave me hanging...!

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 7]
```diff
@@ -0,0 +1,625 @@
...
+    return len(inds[0])
+
+
+def sortSourcesByFlux(sources, reverse=False):
+    """Sort the sources by flux, returning the brightest first.
+
+    Parameters
+    ----------
+    sources : `list` of
+              `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:201]: `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source` -> `lsst.summit.extras.fastStarTrackerAnalysis.Source` ?

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 8]
```diff
@@ -0,0 +1,625 @@
...
+
+    plt.tight_layout()
+
+
+def plotSource(source):
+    """Plot a single source.
+
+    Parameters
+    ----------
+    source : `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source`
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:608]: `lsst.summit.utils.astrometry.fastStarTrackerAnalysis.Source` -> `lsst.summit.extras.fastStarTrackerAnalysis.Source` ?

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py [hunk 9]
```diff
@@ -43,15 +43,18 @@
...
     "countOverThresholdPixels",
     "sortSourcesByFlux",
     "findFastStarTrackerImageSources",
     "checkResultConsistency",
     "plotSourceMovement",
-    "plotSource",
     "plotSourcesOnImage",
+    "plotSource",
+    "Source",
+    "NanSource",
```
Comment (laurenam) [inline python/lsst/summit/extras/fastStarTrackerAnalysis.py:57]: If you're sorting by order of appearance in the file, these last two need to move up above `findFastStarTrackerImageSources`.

**Review by laurenam — APPROVED** [review]

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py (hunk 3 above)
Comment (mfisherlevine) [reply python/lsst/summit/extras/fastStarTrackerAnalysis.py:573]: Needs changing, but to `lsst.summit.extras.fastStarTrackerAnalysis.Source` I think.

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py (hunk 4 above)
Comment (mfisherlevine) [reply python/lsst/summit/extras/fastStarTrackerAnalysis.py:452]: Yup. Done here and everywhere.

File: python/lsst/summit/extras/fastStarTrackerAnalysis.py (hunk 6 above)
Comment (mfisherlevine) [reply python/lsst/summit/extras/fastStarTrackerAnalysis.py]: 🤣

**Pull request merged**